lib/exercises.rb

```diff
@@ -2,19 +2,51 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  hash = {}
+  acc = 0
+  return_array = []
+  strings.each do |str|
+    if !(hash[str.split(//).sort!.join().hash])
+      hash[str.split(//).sort!.join().hash] = acc
+      return_array[acc] = [str]
+      acc += 1
+    else
+      return_array[hash[str.split(//).sort!.join().hash]] << str
+    end
+  end
+  return return_array
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
 # Time Complexity: ?
 # Space Complexity: ?
 def top_k_frequent_elements(list, k)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  # load a hash table which has elements & their frequency 
+  element_frequency = {}
+  array = []
+  return array if list.length == 0
+
+  list.each do |val|
+    if !(element_frequency[val])
+      element_frequency[val] = 1
+    else
+      element_frequency[val] += 1
+    end
+  end
+
+  values = element_frequency.values.sort!
+  inverted_hash = element_frequency.invert
```

Comment:
The only problem here is that if two elements appear the same number of times... one of those elements gets lost.